File: wxDesigner/samples2.19/menu/menu.pl
#!/usr/bin/perl
#----------------------------------------------------------------------------
# Name:         menu.pl
# Author:       XXXX
# Created:      XX/XX/XX
# Copyright:    
#----------------------------------------------------------------------------

use strict;
use Wx;

do 'menu_wdr.pl';

# constants

# WDR: classes

package MyFrame;

use strict;
use vars qw(@ISA);

@ISA=qw(Wx::Frame);

use Wx::Event qw(EVT_MENU EVT_CLOSE EVT_SIZE EVT_UPDATE_UI);
use Wx qw(wxOK wxID_ABOUT wxID_EXIT wxICON_INFORMATION);

sub new {
    my( $class ) = shift;
    my( $this ) = $class->SUPER::new( @_ );

    $this->CreateMyMenuBar();
    
    $this->CreateStatusBar(1);
    $this->SetStatusText("Welcome!", 0);
    
    # insert main window here
    
    # WDR: handler declarations for MyFrame
    EVT_UPDATE_UI( $this, $main::ID_TEST3, \&OnUpdateTest3 );
    # EVT_MENU_HIGHLIGHT( $this, $main::ID_TEST2, \&OnHighlightTest2 );
    EVT_MENU( $this, $main::ID_TEST1, \&OnTest1 );
    EVT_MENU( $this, wxID_ABOUT, \&OnAbout );
    EVT_MENU( $this, wxID_EXIT, \&OnQuit );
    EVT_CLOSE( $this, \&OnCloseWindow );
    
    $this;
}

# WDR: methods for MyFrame

sub CreateMyMenuBar {
    my( $this ) = shift;
    
    $this->SetMenuBar( &main::MyMenuBarFunc() );
}

# WDR: handler implementations for MyFrame

sub OnUpdateTest3 {
    my( $this, $event ) = @_;
    $event->Enable( 0 );
}

sub OnHighlightTest2 {
    my( $this, $event ) = @_;
    $this->SetStatusText( "Test 2 highlighted !!" );
}

sub OnTest1 {
    my( $this, $event ) = @_;
    
    Wx::MessageBox( "Test 1 checked or unchecked",
        "About SuperApp", wxOK|wxICON_INFORMATION, $this );
}

sub OnAbout {
    my( $this, $event ) = @_;
    
    Wx::MessageBox( "Welcome to SuperApp 1.0\n(C)opyright Joe Hacker",
        "About SuperApp", wxOK|wxICON_INFORMATION, $this );
}

sub OnQuit {
    my( $this, $event ) = @_;
    
    $this->Close(1);
}

sub OnCloseWindow {
    my( $this, $event ) = @_;
    
    $this->Destroy();
}

package MyApp;

use strict;
use vars qw(@ISA);

@ISA=qw(Wx::App);

sub OnInit {
    my( $this ) = @_;
    
    Wx::InitAllImageHandlers();
    
    my( $frame ) = MyFrame->new( undef, -1, "SuperApp", [20,20], [500,340] );
    $frame->Show(1);
    
    1;
}

package main;

my( $app ) = MyApp->new();
$app->MainLoop();

File: wxDesigner/samples2.19/menu/menu.py
#!/bin/env python
#----------------------------------------------------------------------------
# Name:         menu.py
# Author:       XXXX
# Created:      XX/XX/XX
# Copyright:    
#----------------------------------------------------------------------------

import wx

from menu_wdr import *


# WDR: classes

class MyFrame(wx.Frame):
    def __init__(self, parent, id, title,
        pos = wx.DefaultPosition, size = wx.DefaultSize,
        style = wx.DEFAULT_FRAME_STYLE ):
        wx.Frame.__init__(self, parent, id, title, pos, size, style)
        
        self.CreateMyMenuBar()
        
        self.CreateStatusBar(1)
        self.SetStatusText("Welcome!")
        
        # insert main window here
        
        # WDR: handler declarations for MyFrame
        wx.EVT_UPDATE_UI(self, ID_TEST3, self.OnUpdateTest3)
        wx.EVT_MENU_HIGHLIGHT(self, ID_TEST2, self.OnHightlightTest2)
        wx.EVT_MENU(self, ID_TEST1, self.OnTest1)
        wx.EVT_MENU(self, wx.ID_ABOUT, self.OnAbout)
        wx.EVT_MENU(self, wx.ID_EXIT, self.OnQuit)
        wx.EVT_CLOSE(self, self.OnCloseWindow)
        
    # WDR: methods for MyFrame
    
    def CreateMyMenuBar(self):
        self.SetMenuBar( MyMenuBarFunc() )
    
    # WDR: handler implementations for MyFrame
    
    def OnUpdateTest3(self, event):
        event.Enable( False );

    def OnHightlightTest2(self, event):
        self.SetStatusText( "Test 2 hightlighted !!" )

    def OnAbout(self, event):
        dialog = wx.MessageDialog(self, "Welcome to SuperApp 1.0\n(C)opyright Joe Hacker",
            "About SuperApp", wx.OK|wx.ICON_INFORMATION )
        dialog.CentreOnParent()
        dialog.ShowModal()
    
    def OnQuit(self, event):
        self.Close(True)
    
    def OnCloseWindow(self, event):
        self.Destroy()
    

#----------------------------------------------------------------------------

class MyApp(wx.App):
    
    def OnInit(self):
        wx.InitAllImageHandlers()
        frame = MyFrame(None, -1, "SuperApp", [20,20], [500,340])
        frame.Show(True)
        return True

#----------------------------------------------------------------------------

app = MyApp(True)
app.MainLoop()

